edkrepo/common/workspace_maintenance/manifest_repos_maintenance.py

```diff
@@ -167,6 +167,10 @@ def find_project_in_all_indices (project, edkrepo_cfg, edkrepo_user_cfg, except_
     edkrepo_user.cfg. If a project with the same name is found uses man_repo to select
     the correct entry
     '''
+
+    # Ensure that up to date copies of all manifest repos are present on the system before searching for a specific
+    # project within them.
+    pull_all_manifest_repos(edkrepo_cfg, edkrepo_user_cfg, reset_hard=False)

```

Comment:
This is going to be a big performance hit. find_project_in_all_indices() calls itself recursively.

Reply:
I have update the PR so that the manifest repo is only cloned if not present.